person.rb

```diff
@@ -1,8 +1,34 @@
-class Person
+class Nameable
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method #{__method__}"
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
```

Comment:
- [ OPTIONAL ] Your Trimmer works good, but I would suggest you use something like this, in order to have an else in case anything unexpected happens:
`@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`